cmd/kubeadm/app/phases/addons/dns/manifests.go

```diff
@@ -257,7 +257,7 @@ spec:
               topologyKey: kubernetes.io/hostname
       containers:
       - name: coredns
-        image: coredns/coredns:{{ .Version }}
+        image: {{ .ImageRepository }}/coredns:{{ .Version }}
```

Comment:
this will change the default to gcr.io... why would we want that?

Reply:
make the kubeadm's config `imageRepository` can implement on coredns.

Reply:
@timothysc

Reply:
We can't switch to gcr.io now. That will have to happen before GA but right now that will just cause a failure since there is no CoreDNS image there yet.

Reply:
@johnbelamaric , Thanks for your reply. As I wrote in **Special notes for your reviewer**, the only problem for this PR is that we need a progress to sync the coreDNS image from `docker hub` to `gcr.io`.  I prefer make `kubeadm` to support overwrite the coreDNS's registry, even it is not GA.

Reply:
From my past discussions with the team, the gcr.io piece will require a fork owned by k8s so that the Kubernetes community has control over the actual image delivered. We should bring this up in the sig-network meeting. The plan was to have that in place by 1.11.

Reply:
But if there is another process we can use to populate grc.io, then this can work.

Reply:
Thank you for your explanation. @johnbelamaric ! You mean the team has discussed about this question and decided to sync coreDNS images to grc.io in `Kubernetes v1.11`?   

Do we have the other way to get new coreDNS images into grc.io?  @timothysc .

Shall we create some mechanisms to handle this situation(break `kubeadm's` behavior) ?

Reply:
> Do we have the other way to get new coreDNS images into grc.io? @timothysc . 

Only a googler can do this /cc @roberthbailey

Reply:
/cc @bowei - is someone from sig-networking shepherding coreDNS images into gcr.io?

Reply:
The transfer of CoreDNS images into grc.io will be part of the finalization for CoreDNS GA.
sig-networking asked to prepare a KEP for this phase.

I am currently working on its preparation. I expect this document to be available next week.
it will be located here : https://github.com/kubernetes/community/tree/master/keps/sig-network

I will reference this PR in the part of the moving generation under control of K8s and hosting the image in gcr.io.